sync/src/main/java/tech/pegasys/artemis/sync/SyncManager.java

```diff
@@ -49,23 +54,41 @@ public SyncManager(
       return CompletableFuture.completedFuture(null);
     }
 
+    syncActive.set(true);
```

Comment:
This should probably use `compareAndSet` and exit immediately if it didn't change from false to true.  Otherwise we could wind up with multiple syncs running.  Currently though that won't work because we sometimes recursively call `executeSync`.  So I think we need to do the syncActive check back up in sync.  So you'd have:

```
public CompletableFuture<Void> sync() {
  if (!syncActive.compareAndSet(false, true)) {
   return CompletableFuture.completedFuture(null); // or maybe should be completed exceptionally because we're already running?
  }
  return executeSync().thenRun(() -> syncActive.set(false));
}
```

and `executeSync` wouldn't touch `syncActive` at all.

Reply:
yeah completing exceptionally in that case makes sense, since we're not expecting to already be in sync when SyncManager.sync() is called